service/serviceutil/serviceutil.go

```diff
@@ -2,15 +2,95 @@ package serviceutil
 
 import (
 	"fmt"
-	"os/exec"
+	"os"
+	"path/filepath"
 	"strings"
 )
 
 // GitRelWorkdir returns git relative workdir of current directory.
+//
+// It should return the same output as `git rev-parse --show-prefix`.
+// It does not execute `git` command to avoid needless git binary dependency.
+//
+// An example problem due to `git` command dependency: https://github.com/reviewdog/reviewdog/issues/1158
 func GitRelWorkdir() (string, error) {
-	b, err := exec.Command("git", "rev-parse", "--show-prefix").Output()
+	cwd, err := os.Getwd()
 	if err != nil {
-		return "", fmt.Errorf("failed to run 'git rev-parse --show-prefix': %w", err)
+		return "", err
 	}
-	return strings.Trim(string(b), "\n"), nil
+	root, err := findGitRoot(cwd)
+	if err != nil {
+		return "", err
+	}
+	if !strings.HasPrefix(cwd, root) {
+		return "", fmt.Errorf("cannot get GitRelWorkdir: cwd=%q, root=%q", cwd, root)
+	}
+	const separator = string(filepath.Separator)
+	path := strings.Trim(strings.TrimPrefix(cwd, root), separator)
+	if path != "" {
+		path = path + separator
+	}
+	return path, nil
+}
+
+func findGitRoot(path string) (string, error) {
+	gitPath, err := findDotGitPath(path)
+	if err != nil {
+		return "", err
+	}
+	return filepath.Dir(gitPath), nil
+}
+
+func findDotGitPath(path string) (string, error) {
+	// normalize the path
+	path, err := filepath.Abs(path)
+	if err != nil {
+		return "", err
+	}
+
+	for {
+		fi, err := os.Stat(filepath.Join(path, ".git"))
+		if err == nil {
+			if !fi.IsDir() {
+				return "", fmt.Errorf(".git exist but is not a directory")
+			}
+			return filepath.Join(path, ".git"), nil
+		}
+		if !os.IsNotExist(err) {
+			// unknown error
+			return "", err
+		}
+
+		// detect bare repo
+		ok, err := isGitDir(path)
+		if err != nil {
+			return "", err
+		}
+		if ok {
+			return path, nil
+		}
+
+		if parent := filepath.Dir(path); parent == path {
+			return "", fmt.Errorf(".git not found")
+		} else {
```

Comment:
**[golint-circleci]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>if block ends with a return statement, so drop this else and outdent its block (move short variable declaration to its own line if necessary)